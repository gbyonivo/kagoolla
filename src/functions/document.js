const doc = document; // eslint-disable-line

export const createElement = (tagName, options = {}, innerHTML = '') => {
  const element = doc.createElement(tagName);
  Object.keys(options).forEach((key) => {
    element.setAttribute(key, options[key]);
  });
  element.innerHTML = innerHTML;
  return element;
};

export const addElementsToElement = (element, elements) => {
  elements.forEach((el) => {
    element.appendChild(el);
  });
};

export const getElementById = id => doc.querySelector(`#${id}`);

export const getElementsByClass = className => doc.querySelectorAll(`.${className}`);
