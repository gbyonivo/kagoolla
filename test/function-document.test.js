import * as documentFunctions from '../src/functions/document';

describe('documentFunctions.createElement', () => {
  it('should create dom', () => {
    const actual = documentFunctions.createElement('li', { class: 'eli' });
    expect(actual).toMatchSnapshot();
  });
  it('should create dom', () => {
    const actual = documentFunctions.createElement('span');
    expect(actual).toMatchSnapshot();
  });
});

describe('documentFunctions.addElementsToElement', () => {
  it('should create dom and add child element to dom', () => {
    const parentElement = documentFunctions.createElement('div', { class: 'eli' });
    const childElement = documentFunctions.createElement('span', { class: 'span' });
    documentFunctions.addElementsToElement(parentElement, [childElement]);
    expect(parentElement).toMatchSnapshot();
  });
});

describe('documentFunctions.getElementById', () => {
  it('should return element created and appended to page', () => {
    const expected = documentFunctions.createElement('section', { id: 'section' });
    document.body.appendChild(expected);
    const actual = documentFunctions.getElementById('section');
    expect(actual).toMatchObject(expected);
  });
});

describe('documentFunctions.getElementByClass', () => {
  it('should return element created and appended to page', () => {
    const expected = documentFunctions.createElement('section', { class: 'section' });
    document.body.appendChild(expected);
    const actual = documentFunctions.getElementsByClass('section');
    expect(actual[0]).toMatchObject(expected);
  });
});