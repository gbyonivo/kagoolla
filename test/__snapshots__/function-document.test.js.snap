// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`documentFunctions.addElementsToElement should create dom and add child element to dom 1`] = `
<div
  class="eli"
>
  <span
    class="span"
  />
</div>
`;

exports[`documentFunctions.createElement should create dom 1`] = `
<li
  class="eli"
/>
`;

exports[`documentFunctions.createElement should create dom 2`] = `<span />`;
